lms/djangoapps/certificates/apis/v0/views.py

```diff
@@ -232,6 +235,21 @@ def get(self, request, username):
 
         return Response(user_certs)
 
+    def _viewable_by_requestor(self, request, username):
+        """
+        Returns whether or not the requesting user is allowed to view the given user's certificates.
+        """
+        try:
+            user = User.objects.get(username=username)
```

Comment:
To prevent extra queries, can we prefetch `Profile` and `Preferences` when we get the user object?

Reply:
`UserProfile` yes. Unfortunately, prefetching `UserPreference`s won't help us here do to the way we fetch them down the stack.